### PR TITLE
Translate Users views to Spanish

### LINKED ISSUE #1576: I18n Spanish: Users
## What are we trying to do?

This ticket is part of the DARIA translation effort. The goal is to implement all our text in both English and Spanish through the i18n* process. 

This ticket is for the User views.  
Code link:  
app/views/users/_search_form.html.erb  
app/views/users/_users_table.html.erb  
app/views/users/edit.html.erb  
app/views/users/index.html.erb  
app/views/users/new.html.erb  
	
The current English text should be moved to `config/locales/en.yml`,   
and new Spanish text should be added to `config/locales/es.yml`.

**Google translate is perfectly acceptable!**  
PRs will be reviewed by a fluent CM to correct any weirdness.

### Steps

For each view  
 - find each instance of displayed text on the view
   `<%= f.date_field :appointment_date, label: 'Appointment date', autocomplete: 'off', help: "Approx gestation at appt: #{patient.last_menstrual_period_at_appt}" %>`
 - replace it with a call to i18n `t` function with a unique identifier based on the file path & field
   `<%= f.date_field :appointment_date, label: t('patient.dashboard.appt_date'), autocomplete: 'off', help: t('patient.dashboard.approx_gestation', gest: patient.last_menstrual_period_at_appt) %>`
 - add a config entry for each language with that identifier as the key, and the language line as the value.
   `en.yml`:
   ```
   patient:
     dashboard:
       approx_gestation: 'Approx gestation at appt: %{gest}'
       appt_date: Appointment date
    ```  
    `es.yml`:
    ```
    patient:
      dashboard:
        approx_gestation: 'Aprox la gestación en la cita: %{gest}'
        appt_date: Día de la cita
    ```

### Completed View Example

For an example of a completed view translation, see [the footer view i18n PR](https://github.com/DCAFEngineering/dcaf_case_management/pull/1547/files).

### Formatting in the YAML files

Each view should have a it's own section. If you see the same key being used across Some commonly used values can go under a shared secton. `app/views/patients/_patient_dashboard.html.erb` keys go under `patient:` -> `dashboard:`.

### Finalizing with i18n-tasks

After you finish all the keys, you should run the i18n checker before committing! This sorts the keys in a standard way and checks for missing keys.

`i18n-tasks normalize`
`i18n-tasks health` - address any issues this highlights!

[More info on using i18n-tasks](https://github.com/glebm/i18n-tasks)

\* 'i18n' is shorthand for 'Internationalization'. [More info on i18n in Rails](https://guides.rubyonrails.org/i18n.html)

Part of #1382

### PR DESCRIPTION
**Pull request by DarthHater**
I rule and have completed some work on Case Manager that's ready for review!

💥 here comes Users in Spanish!

This pull request makes the following changes:

Updates to:
```
app/views/users/_search_form.html.erb
app/views/users/_users_table.html.erb
app/views/users/edit.html.erb
app/views/users/index.html.erb
app/views/users/new.html.erb
```

Adds translations to:
`config/locales/en.yml`
`config/locales/es.yml`

screen shot 2019-02-19 at 8 55 18 am

screen shot 2019-02-19 at 8 55 49 am

It relates to the following issue #s: 
* Fixes #1576

### CODE REVIEW

**Comment by colinxfleming** [comment]
@montanezp

File: config/locales/es.yml [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
   common:
     add: Añadir
+    back: Atrás
     appointment_date_short: Día de la cita
+    email: Email
     name: Nombre
     notes: Notas
     phone: Teléfono
     search: Buscar
+    save: Salvar
```
Comment (montanezp) [inline config/locales/es.yml]: Guardar

File: config/locales/es.yml [hunk 2]
```diff
@@ -205,6 +208,26 @@ es:
...
+    user_table:
+      full_name: Nombre Completo
+      date_created: Fecha de Creacion
+    edit:
+      admin_actions: Acciones de Administrador
+      lock_an_account: Bloquear una Cuenta
+      update_users_role_permissions: Actualizar la función y los permisos de este usuario
+      update_details: Actualizar detalles
+    index:
+      user_account_management: Gestión de Cuentas de Usuario
```
Comment (montanezp) [inline config/locales/es.yml]: Administración de Cuentas de Usuario

Comment (DarthHater) [reply config/locales/es.yml]: In: cb16d37

File: config/locales/es.yml (hunk 1 above)
Comment (DarthHater) [reply config/locales/es.yml]: In: cb16d37

**Pull request merged**